### PR TITLE
Feature/add lesson service tests

### PR DESCRIPTION
**Pull request by SerjKocherga**
Added unit-tests for LessonService methods:
- CreateLessonAsync()
- UpdateLessonAsync()

### CODE REVIEW

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+                MentorId = 2,
+                StudentGroupId = 3,
+                LessonDate = DateTime.Parse("2020-11-18T15:00:00.384Z"),
+                LessonVisits = visitsDto
+            };
+
+            var lessonRepositoryMock = new Mock<ILessonRepository>();
+            lessonRepositoryMock.Setup(x => x.Add(It.IsAny<Lesson>()))
+                .Callback<Lesson>(x => {
+
```
Comment (BohdanBBG) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs:70]: Remove empty line

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+                    Assert.Equal(lessonsDto[i].LessonVisits[j]?.Presence, result[i].LessonVisits[j]?.Presence);
+                    Assert.Equal(lessonsDto[i].LessonVisits[j]?.StudentId, result[i].LessonVisits[j]?.StudentId);
+                    Assert.Equal(lessonsDto[i].LessonVisits[j]?.StudentMark, result[i].LessonVisits[j]?.StudentMark);
+                }
+                
+                Assert.Equal(lessonsDto[i].MentorId, result[i].MentorId);
+                Assert.Equal(lessonsDto[i].StudentGroupId, result[i].StudentGroupId);
+                Assert.Equal(lessonsDto[i].ThemeName, result[i].ThemeName);
+            }
+        }
```
Comment (BohdanBBG) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs:248]: It's necessary? You mock the return GetAll result of the repository, but the method GetAllLessonsAsync of the Lesson service has no other data processing logic, so if you are mock repository GetAll method test is useless

**Review by BohdanBBG — CHANGES_REQUESTED** [review]

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 2 above)
Comment (SerjKocherga) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs:248]: Is it better to remove fully the test for GetAllLessons() ?

Comment (BohdanBBG) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs:248]: Yeap

**Review by BohdanBBG — APPROVED** [review]

**Review by sober607 — APPROVED** [review]

**Review by Vlad1023 — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            Assert.Equal(createdLesson.Id, result.Id);
+            Assert.Equal(createdLesson.LessonDate, result.LessonDate);
+            Assert.Equal(createdLesson.LessonVisits.Count, result.LessonVisits.Count);
+
+            for (int i = 0; i < result.LessonVisits?.Count; i++)
+            {
+                Assert.Equal(createdLesson.LessonVisits[i]?.Comment, result.LessonVisits[i]?.Comment);
+                Assert.Equal(createdLesson.LessonVisits[i]?.Presence, result.LessonVisits[i]?.Presence);
+                Assert.Equal(createdLesson.LessonVisits[i]?.StudentId, result.LessonVisits[i]?.StudentId);
+                Assert.Equal(createdLesson.LessonVisits[i]?.StudentMark, result.LessonVisits[i]?.StudentMark);
```
Comment (x1mx1x) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs:112]: there are too much symbols in the line
you can divide it into several lines

**Review by x1mx1x — APPROVED** [review]

**Review by SnipeFrost — APPROVED** [review]
Not sure is it right to Equal every property

**Pull request merged**